tinygrad/codegen/linearizer.py

```diff
@@ -399,7 +399,6 @@ def calc_tc_idxs(local_size: int, aliases: List[List[int]]):
   def uop(self, uop:UOps, dtype:Optional[DType], vin:Tuple[UOp, ...], arg:Any=None, cachable=True) -> UOp:
     key = (uop, dtype, vin, arg)
     if uop == UOps.PHI and len(vin) == 2 and vin[0] == vin[1]: return vin[0]   # self phi is noop
-    if uop == UOps.CAST and all(x.uop == UOps.GEP for x in vin) and all_same([x.vin[0] for x in vin]) and all(x.arg == i for i,x in enumerate(vin)): return vin[0].vin[0]
```

Comment:
I couldn't understand the original intention behind this - but it doesn't work when we render cast from the AST, eg. if we have 4 GEPs coming from a _float4 -> half cast, the cast and the dtype along with it get thrown away and vin[0].vin[0] ends up being the _float4's LOAD op, I removed this out of necessity to make a kernel like:
```
kernel void E_4(device half* data0, const device float* data1, uint3 gid [[threadgroup_position_in_grid]], uint3 lid [[thread_position_in_threadgroup]]) {
  float4 val0 = (float4)(*((device float4*)(data1+0)));
  *((device half4*)(data0+0)) = (half4)float4((half)((val0).x),(half)((val0).y),(half)((val0).z),(half)((val0).w));
}
```
work

Reply:
Hmm, yea this is sort of wrong. The intention was to force the GPU to broadcast I'm okay with removing.